azurelinuxagent/common/osutil/nsbsd.py

```diff
@@ -0,0 +1,142 @@
+                return
+
+            self.resolver = dns.resolver.Resolver()
+            servers = []
+            cmd = "getconf /usr/Firewall/ConfigFiles/dns Servers | tail -n +2"
+            ret, output = shellutil.run_get_output(cmd)
+            for server in output.split("\n"):
+                if server == '':
+                    break
+                server = server[:-1] # remove last '='
+                cmd = "grep '{}' /etc/hosts".format(server) + " | awk '{print $1}'"
+                ret, ip = shellutil.run_get_output(cmd)
+                servers.append(ip)
+            self.resolver.nameservers = servers
+            dns.resolver.override_system_resolver(self.resolver)
+
+    def set_hostname(self, hostname):
+        shellutil.run("/usr/Firewall/sbin/setconf /usr/Firewall/System/global SystemName {0}".format(hostname))
+        shellutil.run("/usr/Firewall/sbin/enlog")
+        shellutil.run("/usr/Firewall/sbin/enproxy -u")
+        shellutil.run("/usr/Firewall/sbin/ensl -u")
+        shellutil.run("/usr/Firewall/sbin/ennetwork -f")
+
+    def restart_ssh_service(self):
+        return shellutil.run('/usr/Firewall/sbin/enservice', chk_err=False)
+
+    def conf_sshd(self, disable_password):
+        logger.warn("SSHD already configured")
+
+    def useradd(self, username, expiration=None):
+        """
+        Create user account with 'username'
+        """
+        logger.warn("User creation disabled")
+        return
+
+    def del_account(self, username):
+        logger.warn("User deletion disabled")
+
+    def conf_sudoer(self, username, nopasswd=False, remove=False):
+        logger.warn("Sudo is not enabled")
+
+    def chpasswd(self, username, password, crypt_id=6, salt_len=10):
+        cmd = "/usr/Firewall/sbin/fwpasswd -p {0}".format(password)
+        ret, output = shellutil.run_get_output(cmd, log_cmd=False)
+        if ret != 0:
+            raise OSUtilError(("Failed to set password for admin: {0}"
+                               "").format(output))
+
+        # password set, activate webadmin and ssh access
+        shellutil.run('setconf /usr/Firewall/ConfigFiles/webadmin ACL any && ensl',
+                      chk_err=False)
+        shellutil.run('setconf /usr/Firewall/ConfigFiles/system SSH State 1',
+                      chk_err=False)
+        shellutil.run('setconf /usr/Firewall/ConfigFiles/system SSH Password 1',
+                      chk_err=False)
+        shellutil.run('enservice', chk_err=False)
+
+    def del_root_password(self):
+        logger.warn("Root password deletion disabled")
+
+    def start_dhcp_service(self):
+        shellutil.run("/usr/Firewall/sbin/nstart dhclient", chk_err=False)
+
+    def stop_dhcp_service(self):
+        shellutil.run("/usr/Firewall/sbin/nstop dhclient", chk_err=False)
+
+    def get_dhcp_pid(self):
+        ret = shellutil.run_get_output("ps ax | grep dhclient | grep -v grep |  cut -f 1 -d ' '", chk_err=False)
```

Comment:
I would expect the PID to be stored in /var/run/dhclient.pid, but perhaps this has been so customized it isn't?

Reply:
yes, I just translated the pgrep from default.py